include/deal.II/base/parameter_acceptor.h

```diff
@@ -516,6 +520,72 @@ class ParameterAcceptor : public Subscriptor
 
 
 
+/**
+ * A proxy ParameterAcceptor wrapper for classes that have a static member
+ * function @p declare_parameters, and a non virtual @p parse_parameters method.
+ *
+ * If a class already implements the static member function declare_parameters,
+ * and the method declare_parameters, then it cannot be derived directly from
+ * ParameterAcceptor, as the existing member functions would be incompatible
+ * with the virtual methods with the same name that ParameterAcceptor expects
+ * you to overload. For these cases, you can still use ParameterAcceptor
+ * facilities and logic by wrapping your class inside this proxy. The resulting
+ * class will have the public interface of your @p SourceClass, and will provide
+ * the correct interface to ParameterAcceptor.
+ *
+ * An example usage is given by the following snippet of code, using
+ * Functions::ParsedFunction as an example source class:
+ *
+ * @code
+ * ParameterAcceptorProxy<Functions::ParsedFunction<2> > fun("Some function");
+ * ParameterAcceptor::initialize("test.prm");
```

Comment:
I forgot -- does the object created in the first step register itself with a singleton instance that you query in the second line?

Reply:
Yes. ParameterAcceptor has a global `ParameterHandler` member `ParameterAcceptor::prm`, and a global list of classes that register with it. For every class that has been registered, i.e., for every class that was derived by ParameterAcceptor, and that was constructed before the `initialize` function is called, the declare and parse is called with `initialize`.